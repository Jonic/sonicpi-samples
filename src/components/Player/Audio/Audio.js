import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  trackPlay,
  trackPlayBySample,
} from '../../../helpers/Analytics'

class Audio extends Component {
  componentDidUpdate() {
    this.setPlayingState()
  }

  audioSource = () => {
    return `${process.env.PUBLIC_URL}/assets/audio/${this.props.currentSample}.mp3`
  }

  isLooping() {
    if (this.props.isLooping) {
      return 'loop'
    }

    return false
  }

  sampleEnded = () => {
    this.props.sampleEndedHandler()
  }

  setPlayingState = () => {
    if (this.props.isPlaying) {
      return this.audioElement.play()
    }

    this.audioElement.pause()
    this.audioElement.currentTime = 0

    return null
  }

  render() {
    return (
      <div className="c-Audio">
        <audio
          className="c-Audio__element"
          loop={this.isLooping()}
          onEnded={this.sampleEnded}
          ref={
            (audioElement) => {
              this.audioElement = audioElement
            }
          }
          src={this.audioSource()}
        ></audio>
      </div>
    )
  }
}

Audio.propTypes = {
  currentSample:      PropTypes.string.isRequired,
  isLooping:          PropTypes.bool.isRequired,
  isPlaying:          PropTypes.bool.isRequired,
  sampleEndedHandler: PropTypes.func.isRequired,
}

export default Audio
