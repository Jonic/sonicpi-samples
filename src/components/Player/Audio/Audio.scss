@import '../../../styles/component-helper';

.c-audio {}
