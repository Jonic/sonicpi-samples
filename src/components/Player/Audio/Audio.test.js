import 'web-audio-test-api'

import Audio from './Audio'

jest.mock('helpers/Analytics')

let wrapper = shallow(<Audio />)

describe('<Audio />', () => {
  describe('Renders', () => {
    it('without crashing', () => {
      expect(wrapper.find('.c-audio')).toExist()
    })

    it('matches its snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
