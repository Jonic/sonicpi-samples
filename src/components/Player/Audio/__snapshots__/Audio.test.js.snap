// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Audio /> Renders matches its snapshot 1`] = `
<audio
  className="c-audio"
  onEnded={[Function]}
  onPlay={[Function]}
  src={null}
/>
`;
